### PR TITLE
feat: add support for Python3.11.2 version in workspaces

### PR DESCRIPTION
**Pull request by AasthaMehtaTech**
## Description

**This PR is to: add support for Python3.11.2 version in workspaces image**

## Checklist

_Put an `x` in all that apply and explain why in any that don't

- [x] ❓ I have performed a self-review
- [x] 📝 The commit message is clear and descriptive
- [ ] ✅ I have added tests that prove my fix is effective or that my feature works
- [x] 📄 The documentation is up-to-date
- [x] 🔐 The security considerations section in the PR description is complete - **Please do not remove this**

## Dependencies

**Add links to any pull requests or documentation related to this pull request.**

- N/A

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

- [ ] Yes
- [x] No

**If you answered yes, then please explain below.**

## Security Considerations

Are there any security or data concerns to consider?

Please discuss the security implications/considerations relevant to the proposed change.

**These may include:**

* Security-relevant design decisions
* Concerns
* Important discussions
* Implementation-specific guidance and pitfalls
* An outline of risks and threats

- [ ] Yes
- [x] No

**If you answered yes, then please explain below.**

## Types of change

What kind of change does this Pull Request introduce?
_Put an `x` in the boxes that apply_

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 🛠 Adding or updating configuration files, development scripts etc.
- [ ] ♻️ Refactoring (no functional changes, no API changes)
- [ ] 🧹 Chore (removing redundant files, fixing typos etc.)
- [ ] 📄 Documentation Update
- [ ] ❓ Other (if none of the other choices applies)

**If you answered "other", then please explain below.**

## Additional information

Please add any additional information that will be useful for the reviewer.
- N/A

### CODE REVIEW

File: workspace/python/ds-pack/base/3-11/Dockerfile
Comment (saurabh0402) [inline workspace/python/ds-pack/base/3-11/Dockerfile:1]: Hey, any reason we have opted for this folder structure - `base/3-11`? Why are we creating that `base` folder?
I feel that we should just add the `version` name as the folder name. So, something like `3-11-base` or `3.11-base`.

We currently have not migrated the global images to versioning, but when we do we will have an image called `workspace-python-ds-pack` and within it, we will have many versions. The name of the folder here should correspond to the version name inside the image which can be something like `<semver>-<text>`.

File: workspace/python/ds-pack/base/3-11/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+# SPARK: copy over spark config
+COPY ./spark /usr/local/spark
+
+# VS Code
+COPY codeserver-proxy codeserver-proxy
+RUN curl -fsSL https://code-server.dev/install.sh \
+    | sh -s -- --version 4.11.0 \
+    && pip install --no-cache-dir ./codeserver-proxy
+
```
[the remark this answers is not in the document]
Comment (saurabh0402) [reply workspace/python/ds-pack/base/3-11/Dockerfile:87]: Also, can you please add the error we were getting when that was added?

[the remark this answers is not in the document]
Comment (AasthaMehtaTech) [reply workspace/python/ds-pack/base/3-11/Dockerfile:87]: https://github.com/JetBrains/projector-installer/blob/master/projector_installer/products.py

```
#17 49.54   File "/usr/local/bin/projector", line 5, in <module>
#17 49.55     from projector_installer.cmd import projector
#17 49.55   File "/usr/local/lib/python3.11/site-packages/projector_installer/cmd.py", line 16, in <module>
#17 49.55     from .actions import do_install_app, do_uninstall_app, do_find_app, do_list_app, do_run_config, \
#17 49.55   File "/usr/local/lib/python3.11/site-packages/projector_installer/actions.py", line 23, in <module>
#17 49.55     from .ide_update import is_updatable_ide, get_update, update_config, check_ide_update, is_tested_ide
#17 49.55   File "/usr/local/lib/python3.11/site-packages/projector_installer/ide_update.py", line 12, in <module>
#17 49.55     from .products import Product, get_product_releases, IDEKind, load_compatible_apps, \
#17 49.55   File "/usr/local/lib/python3.11/site-packages/projector_installer/products.py", line 46, in <module>
#17 49.55     @dataclass
#17 49.55      ^^^^^^^^^
#17 49.55   File "/usr/local/lib/python3.11/dataclasses.py", line 1220, in dataclass
#17 49.55     return wrap(cls)
#17 49.55            ^^^^^^^^^
#17 49.55   File "/usr/local/lib/python3.11/dataclasses.py", line 1210, in wrap
#17 49.57     return _process_class(cls, init, repr, eq, order, unsafe_hash,
#17 49.57            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#17 49.57   File "/usr/local/lib/python3.11/dataclasses.py", line 958, in _process_class
#17 49.57     cls_fields.append(_get_field(cls, name, type, kw_only))
#17 49.57                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#17 49.57   File "/usr/local/lib/python3.11/dataclasses.py", line 815, in _get_field
#17 49.57     raise ValueError(f'mutable default {type(f.default)} for field '
#17 49.57 ValueError: mutable default <class 'distutils.version.LooseVersion'> for field ver is not allowed: use default_factory
------
executor failed running [/bin/bash -o pipefail -c pip install --no-cache-dir projector-installer==1.8.0 --upgrade     && projector --accept-license --config-directory /usr/local/share/projector     ide autoinstall --config-name pycharm --ide-name "PyCharm Community Edition 2021.3.1"     && chmod -R a=rwx /usr/local/share/projector     && pip install --no-cache-dir ./projector-proxy]: exit code: 1``
```

File: workspace/python/ds-pack/base/3-11/Dockerfile
Comment (AasthaMehtaTech) [reply workspace/python/ds-pack/base/3-11/Dockerfile:1]: Restructured and named as per the discussion to follow semantic image versioning suffixed with additional base image info

File: workspace/python/ds-pack/2.0.0-base-python-3.11.2/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Linux Kernel   5.10.176
+Python         3.11.2
+```
+
+### Important Linux packages installed
+```
+aws-cli        2.7.20
+curl           7.74.0
+git            2.30.2
+jq             1.6-2.1
```
Comment (saurabh0402) [inline workspace/python/ds-pack/2.0.0-base-python-3.11.2/README.md:22]: I get only `1.6` when I run `jq --help`.

Comment (saurabh0402) [reply workspace/python/ds-pack/2.0.0-base-python-3.11.2/README.md:22]: For nano, its `5.4`

Comment (AasthaMehtaTech) [reply workspace/python/ds-pack/2.0.0-base-python-3.11.2/README.md:22]: jq/now 1.6-2.1 amd64 [installed,local]
nano/now 5.4-2+deb11u2 amd64 [installed,local]

Can see similar patterns for vim, pandoc, etc in Image build logs:  `vim            8.2.2434-3`

Comment (saurabh0402) [reply workspace/python/ds-pack/2.0.0-base-python-3.11.2/README.md:22]: Ahh, I tried running directly on the workspace 😅

**Review by saurabh0402 — APPROVED** [review]

**Pull request merged**